sgx/report-cache/untrusted/src/lib.rs

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+//! The untrusted side of enclave report caching.
+
+use displaydoc::Display;
+use mc_attest_core::{
+    IasQuoteError, PibError, ProviderId, QuoteError, QuoteSignType, TargetInfoError,
+    VerificationReport, VerificationReportData, VerifyError,
+};
+use mc_attest_enclave_api::Error as AttestEnclaveError;
+use mc_attest_net::{Error as RaError, RaClient};
+use mc_attest_untrusted::QuotingEnclave;
+use mc_common::logger::{log, o, Logger};
+use mc_sgx_report_cache_api::{Error as ReportableEnclaveError, ReportableEnclave};
+use mc_util_metrics::IntGauge;
+use retry::{delay::Fibonacci, retry, Error as RetryError, OperationResult};
+use std::{
+    convert::TryFrom,
+    io::Error as IOError,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    thread::{sleep, Builder as ThreadBuilder, JoinHandle},
+    time::{Duration, Instant},
+};
+
+/// How long to wait between report refreshes.
+pub const REPORT_REFRESH_INTERNAL: Duration = Duration::from_secs(18 * 60 * 60); // 18 hours.
```

Comment:
The value of 18 hours was chosen after a brief conversation with @jcape. This can also be made configurable via command line, if there is a need for that.

Reply:
Can you provide a brief summary of the reasoning for 18 hours?

Reply:
I can't :) but maybe @jcape can?